source/neuropod/internal/neuropod_tensor.cc

```diff
@@ -48,7 +48,7 @@ std::vector<int64_t> compute_strides(const std::vector<int64_t> &dims)
 size_t compute_num_elements(const std::vector<int64_t> &dims)
 {
     // Get the number of elements in the tensor by multiplying all the dims together
-    return std::accumulate(dims.begin(), dims.end(), 1, std::multiplies<>());
+    return static_cast<size_t>(std::accumulate(dims.begin(), dims.end(), 1, std::multiplies<>()));
```

Comment:
I remember we had negative dims (-2 for symbol shapes). At this level, is it possible that we have valid negative dim?

Reply:
`compute_num_elements` is used on tensors (not tensor specs) so it is not valid to have negative dims at this point